hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -409,8 +409,11 @@ public static HoodieData<HoodieRecord> convertMetadataToBloomFilterRecords(
         LOG.error("Failed to find path in write stat to update metadata table " + hoodieWriteStat);
         return Collections.emptyListIterator();
       }
-      int offset = partition.equals(NON_PARTITIONED_NAME) ? (pathWithPartition.startsWith("/") ? 1 : 0) :
-          partition.length() + 1;
+
+      // For partitioned table, "partition" contains the relative partition path;
+      // for non-partitioned table, "partition" is empty
+      int offset = StringUtils.isNullOrEmpty(partition)
```

Comment:
Fetching a file name from full path and partition path, should a helper on FSUtils? Move this logic there? better yet, Can't we just String.replace the first occurrence of partition path within the full path and we are done without the index/offset business?

Reply:
Addressed in #6250.  `String.replace` could be slow so I still use the current logic.  I moved it into a util method.